var/spack/repos/builtin/packages/sherpa/package.py

```diff
@@ -100,30 +111,48 @@ def configure_args(self):
         args.append('--enable-hepevtsize=200000')
         args.append('--with-sqlite3=' + self.spec['sqlite'].prefix)
         args.extend(self.enable_or_disable('mpi'))
-        if self.spec.satisfies('+python'):
-            args.append('--enable-pyext')
+        args.extend(self.enable_or_disable('pyext', variant='python'))
         args.extend(self.enable_or_disable('analysis'))
         args.extend(self.enable_or_disable('lhole'))
         args.extend(self.enable_or_disable('gzip'))
         args.extend(self.enable_or_disable('pythia'))
-        if self.spec.satisfies('+hepmc2'):
-            args.append('--enable-hepmc2=' + self.spec['hepmc'].prefix)
-        if self.spec.satisfies('+hepmc3'):
-            args.append('--enable-hepmc3=' + self.spec['hepmc3'].prefix)
-        if self.spec.satisfies('+rivet'):
-            args.append('--enable-rivet=' + self.spec['rivet'].prefix)
-        if self.spec.satisfies('+fastjet'):
-            args.append('--enable-fastjet=' + self.spec['fastjet'].prefix)
-        if self.spec.satisfies('+openloops'):
-            args.append('--enable-openloops=' + self.spec['openloops'].prefix)
-        if self.spec.satisfies('+recola'):
-            args.append('--enable-recola=' + self.spec['recola'].prefix)
-        if self.spec.satisfies('+root'):
-            args.append('--enable-root=' + self.spec['root'].prefix)
-        if self.spec.satisfies('+lhapdf'):
-            args.append('--enable-lhapdf=' + self.spec['lhapdf'].prefix)
-        if self.spec.satisfies('+hztool'):
-            args.append('--enable-hztool=' + self.spec['hztool'].prefix)
-        if self.spec.satisfies('+cernlib'):
-            args.append('--enable-cernlib=' + self.spec['cernlib'].prefix)
+        hepmc_root = lambda x: self.spec['hepmc'].prefix
+        args.extend(self.enable_or_disable('hepmc2', activation_value=hepmc_root))
+        if self.spec.satisfies('@2.2.13:'):
+            args.extend(self.enable_or_disable('hepmc3', activation_value='prefix'))
+            args.extend(self.enable_or_disable('rivet', activation_value='prefix'))
+            args.extend(self.enable_or_disable('lhapdf', activation_value='prefix'))
+        else:
+            # See https://gitlab.com/sherpa-team/sherpa/-/issues/348
+            if self.spec.satisfies('+hepmc3'):
+                args.append('--enable-hepmc3=' + self.spec['hepmc3'].prefix)
+            if self.spec.satisfies('+rivet'):
+                args.append('--enable-rivet=' + self.spec['rivet'].prefix)
+            if self.spec.satisfies('+lhapdf'):
+                args.append('--enable-lhapdf=' + self.spec['lhapdf'].prefix)
```

Comment:
Is the failure just when you specify `--disable-foo`, or does the code fail if any of these externals are missing?

Reply:
`configure` fails if you specify `--disable-hepmc3` (same for rivet, lhapdf) - it still looks for hepmc3 (rivet, lhapdf). The developers have fixed the issue in `master`, will be included in the next version.

Reply:
My question is do they fail if not found? Does the configure error out if disabling or does it simply ignore the flag? If the latter then you don't need the extra logic I think

Reply:
And if it still looks for them and fails if they're unspecified, then you need a conflict instead of the logic

Reply:
@sethrj sorry, I don't quite understand your question. Rivet, LHAPDF and HepMC3 are all optional dependencies. They don't conflict with each other. In Sherpa <= 2.2.12, the `--disable-lhapdf`, `--disable-rivet` and `--disable-hepmc3` options did not work correctly: `--disable-X` did not skip the check for `X`. This is fixed in Sherpa master and will be included in the next version. `--enable-X` options work as expected. I can remove the logic and just not pass `--enable-X` if variant X is not enabled, this will work with all Sherpa versions.

Reply:
> --disable-X did not skip the check for X, while still disabling X
That's what I was looking for: it means your implementation is good as is. Sorry for the confusion!